src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp

```diff
@@ -3902,16 +3902,18 @@ CTranslatorExprToDXL::PdxlnCorrelatedNLJoin(
 	COperator::EOperatorId op_id = pexpr->Pop()->Eopid();
 	CDXLNode *pdxlnCond = nullptr;
 
-	// Create a subplan with a Boolean from the inner child if we have a Const True as a join condition.
-	// One scenario for this is when IN sublinks contain a projection from the outer table only such as:
-	// select * from foo where foo.a in (select foo.b from bar);
-	// If bar is a very small table, ORCA generates a CorrelatedInLeftSemiNLJoin with a Const true join filter
-	// and condition foo.a = foo.b is added as a filter on the table scan of foo. If bar is a large table,
-	// ORCA generates a plan with CorrelatedInnerNLJoin with a Const true join filter and a LIMIT over the
-	// scan of bar. The same foo.a = foo.b condition is also added as a filter on the table scan of foo.
+	// Create a subplan with a Boolean from the inner child if we have a Const
+	// True as a join condition.
+	// One scenario for this is when sublink contain a projection from the outer
+	// table only such as:
+	// select * from foo where 10 > (select bar.b from bar where foo.b = bar.b);
+	// If bar is a very small table, ORCA generates a CorrelatedInnerNLJoin with
+	// a Const true join filter and condition foo.a = foo.b is added as a filter
+	// on the table scan of bar.
+	// For the other cases function BuildSubplans generates another sublink type
+	// for correct type of join condition.
 	if (CUtils::FScalarConstTrue(pexprScalar) &&
-		(COperator::EopPhysicalCorrelatedInnerNLJoin == op_id ||
-		 COperator::EopPhysicalCorrelatedInLeftSemiNLJoin == op_id))
+		COperator::EopPhysicalCorrelatedInnerNLJoin == op_id)
```

Comment:
Why is it correct that subplan returns only one row in the case of correlated inner nested loop join (EopPhysicalCorrelatedInnerNLJoin)?

Reply:
Correlated inner nested loop join is not enough, there must be bool expression, which is checked at the first line in the "if" clause.

Reply:
The fixed problem was that subplan returned more than one tuple. Why will subplan return only one tuple when this condition is true?

Reply:
No, the fixed problem was incorrect subplan, which must return one row, but return more than one row. Now it also returns more than one row and it is expected behavior.